src/main/java/org/mifos/connector/ams/interop/AmsCommonService.java

```diff
@@ -92,6 +98,25 @@ public void repayLoan(Exchange e) {
         headers.put("Content-Type", APPLICATION_TYPE);
         headers.putAll(tenantService.getHeaders(e.getProperty(TENANT_ID, String.class)));
         cxfrsUtil.sendInOut("cxfrs:bean:ams.local.loan", e, headers, e.getIn().getBody());
+//        e.setProperty(FINERACT_RESPONSE_BODY,e.getOut().getBody().toString());
+    }
+    public void sendCallback(Exchange e) {
```

Comment:
make this a generic method.
accept individual variables instead of exchange. also check if this can be asynchronous.

Reply:
changed the signature